internal/controllers/scheduling/controller_test.go

```diff
@@ -469,32 +469,50 @@ func TestDispatchOrder(t *testing.T) {
 	synth := &apiv1.Synthesizer{}
 	synth.Name = "test-synth"
 	synth.Namespace = "default"
+	synth.Generation = 2
 	require.NoError(t, cli.Create(ctx, synth))
 
 	// Waiting for the new synth
 	comp := &apiv1.Composition{}
-	comp.Name = "test-comp-1"
 	comp.Namespace = "default"
 	comp.Finalizers = []string{"eno.azure.io/cleanup"}
 	comp.Generation = 2
 	comp.Spec.Synthesizer.Name = synth.Name
-	comp.Status.CurrentSynthesis = &apiv1.Synthesis{UUID: "foo", ObservedCompositionGeneration: 1, Synthesized: ptr.To(metav1.Now())}
+	comp.Status.CurrentSynthesis = &apiv1.Synthesis{
+		UUID:                          "foo",
+		ObservedCompositionGeneration: comp.Generation,
+		ObservedSynthesizerGeneration: synth.Generation,
+		Synthesized:                   ptr.To(metav1.Now()),
+	}
+
+	comp1 := comp.DeepCopy()
+	comp1.Name = "test-comp-1"
+	comp1.Status.CurrentSynthesis.ObservedCompositionGeneration--
+	require.NoError(t, cli.Create(ctx, comp1))
+	require.NoError(t, cli.Status().Update(ctx, comp1))
 
 	comp2 := comp.DeepCopy()
 	comp2.Name = "test-comp-2"
-	comp2.Status.CurrentSynthesis.ObservedSynthesizerGeneration = synth.Generation
-	require.NoError(t, cli.Create(ctx, comp))
+	comp2.Status.CurrentSynthesis.ObservedSynthesizerGeneration--
```

Comment:
The logic of this test was subtly wrong before which is how the sorting bug was able to merge.